### ISSUE TITLE
Part of prompt applied from a style is not extracted from PNG metada

### ISSUE THREAD

**Issue by satanosskm** [thundaga/SD-webui-txt2img-script#24]
Hi !

I noticed I had completely different pictures when using HiresFix in a batch. Positive, Negative prompts, Seed etc... are checked.

When comparing the PNG metada (from PNG info tab) I noticed that the part of the negative prompt that are applied by a style (all my negative embeddings) are missing from the result hiresfixed picture, but they are found in the original pic.

So the metadata of the PNG are read differently from your script and from the PNG Info tab, and parts of prompts applied by a style are missing, resulting in different pictures.

When applying the style by my own it works as intended, but we have to check if every PNG used applied-style or not.

Congrats for the scrip BTW, really useful.

**Comment by satanosskm** [comment]
The style :
![image](https://github.com/thundaga/SD-webui-txt2img-script/assets/113078026/c322ec82-bb3e-4808-bfb2-1e8f16cf1e90)
PNG info (from the tab), vanilla generation without the script, style applied :
![image](https://github.com/thundaga/SD-webui-txt2img-script/assets/113078026/67f3af8f-eba1-4bf8-83eb-606909b2b404)
Settings from the script for next generation :
![image](https://github.com/thundaga/SD-webui-txt2img-script/assets/113078026/8cb8a694-98b1-4ea4-a421-b1438abd59b0)
PNG info (from the tab), generation with the script, style unapplied + HiresFix enabled :
![image](https://github.com/thundaga/SD-webui-txt2img-script/assets/113078026/69ff7033-cd06-4a43-b63e-8cf6532b2fac)

BTW it's not the HirexFix causing this, the results are similar without it (different pictures, part of the prompt from a style not included).

**Comment by OPCOXE** [comment]
Thanks we really didn't know about that #14

**Comment by satanosskm** [comment]
Yes, it works when imported from PNG Info, that's the way I HiresFix my picture usuallyn because it automatically applies the style.

The fix would be a new setting "style" to read that part of metadata (and choice to apply or not), because it seems apart from "prompt".
